### PR TITLE
CCIP-3785 fix data availability bytes length calc

### PR DESCRIPTION
**Pull request by 0xsuryansh**
PR fixing comments and fixed bytes overhead calculation in `Internal.sol`

### CODE REVIEW

File: contracts/src/v0.8/ccip/libraries/Internal.sol [hunk 1]
```diff
@@ -75,21 +75,25 @@ library Internal {
...
-  uint256 public constant MESSAGE_FIXED_BYTES_PER_TOKEN = 32 * ((2 * 3) + 3);
+  /// 0x20
+  /// destTokenAddress
+  /// destGasAmount
+  /// amount
+  /// sourcePoolAddress_offset
+  /// extraData_offset
+  /// sourcePoolAddress_length
+  /// sourcePoolAddress_content // assume 1 slot
+  /// extraData_length // contents billed separately
```
Comment (RayXpub) [inline contracts/src/v0.8/ccip/libraries/Internal.sol:96]: nit: I think if we follow the struct layout the order would be:
```
/// 0x20
/// sourcePoolAddress_offset
/// destTokenAddress
/// destGasAmount
/// extraData_offset
/// amount
/// sourcePoolAddress_length
/// sourcePoolAddress_content // assume 1 slot
/// extraData_length // contents billed separately
```

File: contracts/src/v0.8/ccip/libraries/Internal.sol [hunk 2]
```diff
@@ -75,21 +75,25 @@ library Internal {
...
   /// Each variable array takes 1 more slot to store its length.
   /// When abi encoded, excluding array contents,
   /// Any2EVMMessage takes up a fixed number of 13 slots, 32 bytes each.
+  /// Assume 1 slot for sender
   /// For structs that contain arrays, 1 more slot is added to the front, reaching a total of 14.
   /// The fixed bytes does not cover struct data (this is represented by MESSAGE_FIXED_BYTES_PER_TOKEN)
-  uint256 public constant MESSAGE_FIXED_BYTES = 32 * 14;
+  uint256 public constant MESSAGE_FIXED_BYTES = 32 * 15;
 
-  /// @dev Each token transfer adds 1 RampTokenAmount
```
Comment (RayXpub) [inline contracts/src/v0.8/ccip/libraries/Internal.sol:86]: nit: I think this comment is useful to mention which struct we are referring to (needs to be updated to `Any2EVMTokenTransfer`)

Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/libraries/Internal.sol:86]: This is a deleted line, updating in the follow up comments

File: contracts/src/v0.8/ccip/libraries/Internal.sol (hunk 1 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/libraries/Internal.sol:96]: I wrote it in terms of how we group the slots and use 4, 3 and 2 in the following calculation. Let me know

Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/libraries/Internal.sol:96]: Making the correction

**Review by RensR — DISMISSED** [review]

**Review by RayXpub — APPROVED** [review]

**Review by RensR — APPROVED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Pull request merged**